### PR TITLE
BUG-116164 Allow multiple components to trigger the same storage location, update cloud storage locations

### PR DESCRIPTION
**Pull request by sidseth**
Critical fix to allow launching clusters via the UI: Required right now since  RMP-1267  and   BUG-114429 are not being addressed this week.
Tested locally to match: https://docs.google.com/spreadsheets/d/10mB1_O4aWbEwrNC5FriCyzvzTJbsJoj-reiAW8DgnNs/edit#gid=0

### CODE REVIEW

**Comment by sidseth** [comment]
The build seems to have failed during the test run (works locally). @akanto, @doktoric  - could you please re-trigger it. I don't have the required permissions on the circleci page.

File: template-manager-blueprint/src/main/resources/definitions/cloud-storage-location-specification.json [hunk 1]
```diff
@@ -3,140 +3,110 @@
     {
       "propertyName": "logfeeder.cloud.storage.base.path",
       "propertyFile": "logfeeder-properties",
-      "description": "This is the directory where cluster daemon logs should be stored by Log Feeder",
+      "description": "This is the directory where cluster daemon logs will be stored by Log Feeder",
       "defaultPath": "{{{ defaultPath }}}/clusterlogs",
```
Comment (lajosrodek) [inline template-manager-blueprint/src/main/resources/definitions/cloud-storage-location-specification.json]: I know I screwed this up but could you please insert `/oplogs` before `/clusterlogs`?

File: template-manager-blueprint/src/main/resources/definitions/cloud-storage-location-specification.json [hunk 2]
```diff
@@ -3,140 +3,110 @@
...
       "requiredForAttachedCluster": false,
       "supportedStorages": ["ADLS", "WASB", "S3", "GCS", "ADLS_GEN_2"]
     },
     {
       "propertyName": "tez.history.logging.proto-base-dir",
       "propertyFile": "tez-site",
-      "description": "The directory into which Tez history data will be written when proto history logging service is used",
-      "defaultPath": "{{{ defaultPath }}}/apps/hive/ewarehouse/tablespace/external/hive/sys.db/query_data",
+      "description": "The directory into which Tez/Hive job history data will be written",
+      "defaultPath": "{{{ defaultPath }}}/warehouse/tablespace/external/hive/sys.db/query_data",
```
Comment (lajosrodek) [inline template-manager-blueprint/src/main/resources/definitions/cloud-storage-location-specification.json]: Please note that due to BUG-115324 `{{{ defaultPath }}}` now never contains the cluster name. If that is required anywhere for the demo, `/{{{ clusterName }}}` or `{{{#if-false attachedCluster}}}/{{{ clusterName }}}{{{/if-false}}}` should be added as appropriate. Sorry for the obfuscation.

**Comment by sidseth** [comment]
Updated PR to differentiate between standalone, attached and datalakes. Also fixed the checkstyle warnings.

File: template-manager-blueprint/src/main/resources/definitions/cloud-storage-location-specification.json (hunk 1 above)
Comment (lajosrodek) [reply template-manager-blueprint/src/main/resources/definitions/cloud-storage-location-specification.json]: Perfect, thanks!

File: template-manager-blueprint/src/main/resources/definitions/cloud-storage-location-specification.json (hunk 2 above)
Comment (lajosrodek) [reply template-manager-blueprint/src/main/resources/definitions/cloud-storage-location-specification.json]: Looks fine. Nice job!

**Comment by lajosrodek** [comment]
@sidseth here are the checkstyle errors for your convenience:
```
> Task :template-manager-blueprint:checkstyleTest FAILED
[ant:checkstyle] [ERROR] /home/circleci/project/template-manager-blueprint/src/test/java/com/sequenceiq/cloudbreak/blueprint/filesystem/query/FileSystemConfigQueryServiceTest.java:298: Line is longer than 161 characters (found 167). [LineLength]
[ant:checkstyle] [ERROR] /home/circleci/project/template-manager-blueprint/src/test/java/com/sequenceiq/cloudbreak/blueprint/filesystem/query/FileSystemConfigQueryServiceTest.java:299: Line is longer than 161 characters (found 166). [LineLength]
[ant:checkstyle] [ERROR] /home/circleci/project/template-manager-blueprint/src/test/java/com/sequenceiq/cloudbreak/blueprint/filesystem/query/FileSystemConfigQueryServiceTest.java:393: Line is longer than 161 characters (found 190). [LineLength]
```

**Pull request merged**